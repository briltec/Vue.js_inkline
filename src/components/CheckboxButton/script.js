import Checkbox from '../Checkbox';

export default {
    name: 'CheckboxButton',
    extends: Checkbox
};
