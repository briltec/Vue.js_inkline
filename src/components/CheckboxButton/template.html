<i-button class="-checkbox"
          :class="classes"
          :active="checked"
          @click="clickInputRef"
          @blur="onBlur"
          @focus="onFocus"
          :aria-disabled="isDisabled"
          :aria-readonly="isReadonly"
          :readonly="isReadonly"
          :tabindex="tabIndex"
          @keydown.space.stop.prevent="clickInputRef">
    <input
        type="checkbox"
        ref="input"
        v-model="model"
        :name="name"
        :value="currentValue"
        :aria-disabled="isDisabled"
        :aria-readonly="isReadonly"
        :readonly="isReadonly"
        :disabled="isDisabled">
    <slot></slot>
</i-button>
