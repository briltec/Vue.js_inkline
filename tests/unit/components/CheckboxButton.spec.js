import { mount } from '@vue/test-utils';
import CheckboxButton from 'inkline/components/CheckboxButton';

describe('Components', () => {
    describe('CheckboxButton', () => {
        let wrapper;

        beforeEach(() => {
            wrapper = mount(CheckboxButton);
        });

        it('should render correctly', () => {
            expect(wrapper.html()).toMatchSnapshot();
        });
    });
});
