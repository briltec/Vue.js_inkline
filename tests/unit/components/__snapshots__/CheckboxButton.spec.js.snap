// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components CheckboxButton should render correctly 1`] = `<i-button-stub activeclass="-active" active="true" tag="button" size="" tabindex="1" variant="" class="-checkbox -custom"><input type="checkbox" tabindex="-1" value="false"> </i-button-stub>`;
